https://issues.redhat.com/browse/ACM-17665--Add Provisioning section to Edge Manager (#7608)
open-cluster-management/rhacm-docs: edge_manager/edge_mgr_provisioning_vmware_vsphere.adoc

[#edge-provisioning-vmware-vsphere-prereq]
== Prerequisites

* You installed the `flightctl` CLI and logged in to your {rhem} service instance.
* You installed the VMware `govc` CLI and logged in to your remote vSphere environment.
* You built a VMDK disk image called `disk.vmdk` and the image is available on the machine running the commands.

[#edge-provisioning-vmware-vsphere-proc]
== Provisioning on VMware vSphere

To create the `cloud-init` configuration, complete the following steps:

. Request a new {rhem} agent enrollment configuration and store it in a file called `config.yaml`. Run the following command:

+
[source,bash]
----
flightctl certificate request --signer=enrollment --expiration=365d --output=embedded > config.yaml
----

. Create a cloud configuration user data file called `cloud-config.yaml` that places the agent configuration in the correct location on the first boot. Run the following command:

+
[source,bash]
----
cat << END > cloud-config.yaml
#cloud-config
write_files:
- path: /etc/flightctl/config.yaml
  content: $(cat config.yaml | base64 -w0)
  encoding: b64

> END

Reviewer: Could you please replace `<< END` with `<<EOF` (no space) on line 33 and `END` with `EOF` on line 39? Either should work, but Ken reported only EOF worked for him.

Author: We have instances of `END` in the Virtualization section too. Should I update them too?

Reviewer: Yes, please.

----

> . Upload the `disk.vmdk` file to the desired directory in vSphere. Run the following command:

Reviewer: ```suggestion
. Upload the `disk.vmdk` file to the desired folder in vSphere. Run the following command:
```
Later we're using the term "folder" instead of "directory". As we're using the variablre `${DESTINATION_FOLDER}`,  I'd suggest replacing "directory" here and below.

Author: Replaced all instances that refers to the `DESTINATION_FOLDER`

+
[source,bash]
----
govc import.vmdk \
  -dc="${DATACENTER}" -ds="${DATASTORE}" -pool="${DATACENTER_POOL}" \
  disk.vmdk ${DESTINATION_FOLDER}
----

> . Confirm that the file was uploaded by listing files within the `datastore` directory by running the following command:

Reviewer: ```suggestion
. Confirm that the file was uploaded by listing files inside the desired folder within the desired datastore by running the following command:
```
There isn't a directory called "datastore", but the user specified a datastore, then a directory (sorry: folder) within that datastore.

+
[source,bash]
----
govc datastore.ls \
  -dc="${DATACENTER}" -ds="${DATASTORE}" \
  ${DESTINATION_FOLDER}
----

+

> If the uploading the file was successful, you see the `disk.vmdk` and `disk-flat.vmdk` files.

Reviewer: ```suggestion
If uploading the file was successful, you see the `disk.vmdk` and `disk-flat.vmdk` files.
```
